src/storage/store/engine.rs

```diff
@@ -947,15 +950,55 @@ impl ShardEngine {
                 // Validate ens
```

Comment:
What about this one?

Reply:
this was pulled out because validation needs to happen at a higher level. did you want there to be just basic field info validation (minus the signature/address?)